### PR TITLE
Patient listings show wrong age

### PR DESCRIPTION
**Pull request by xispa**
## Description of the issue/feature this PR addresses

Linked issue: https://github.com/senaite/senaite.health/issues/160

## Current behavior before PR

Patient listings shows wrong age

## Desired behavior after PR is merged

Patient listings shows correct age

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

File: bika/health/utils.py [hunk 1]
```diff
@@ -163,3 +167,53 @@ def handle_after_submit(context, request, state):
...
+    if not date_value:
+        if default is None:
+            return None
+        return to_datetime(default, tzinfo=tzinfo)
+
+    # Convert to datetime and strip
+    return DT2dt(date_value).replace(tzinfo=tzinfo)
+
+
+def to_ymd(delta):
```
Comment (Espurna) [inline bika/health/utils.py:210]: Shouldn't the date format depend on the region?

File: bika/health/utils.py [hunk 2]
```diff
@@ -163,3 +167,53 @@ def handle_after_submit(context, request, state):
     else:
         status_id = "success"
     return status_id
+
+
+def get_age_ymd(birth_date, to_date=None):
+    """Returns the age at to_date if not None. Otherwise, current age
+    """
+    delta = get_relative_delta(birth_date, to_date)
```
Comment (Espurna) [inline bika/health/utils.py:175]: If a patient does not have birthdate due to an old inconsistency, the system rises an "internal error" to the user without pointing out the source of the issue.

I think that the upgrade step should consider and fix patients without birth date.

File: bika/health/utils.py (hunk 1 above)
Comment (xispa) [reply bika/health/utils.py:210]: This function is not about the date format. It receives a timedelta (difference between two dates) and returns a human-redable string like "13a 2m 1d". It is mainly used to display the age of a patient, not the date of birth.

File: bika/health/utils.py (hunk 2 above)
Comment (xispa) [reply bika/health/utils.py:175]: The Birth Date is a required field in senaite.health:
https://github.com/senaite/senaite.health/blob/master/bika/health/content/patient.py#L133
I think this is the correct behavior

Comment (Espurna) [reply bika/health/utils.py:175]: Right

**Pull request merged**